app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -10,7 +10,7 @@ def determine_placement
   end
 
   def prepare_volumes
-    if options[:volumes]
+    if options[:volumes] && options[:volumes].any?
```

Comment:
Try just `options[:volumes].present?`  that will work with `nil` and `[]`

Reply:
Ah, good point, thanks.